### PR TITLE
File object

### PR DESCRIPTION
**Pull request by WyriHaximus**
This is the first PR extracted from #41 and is solely for the `File` object used in the streaming body parser's. (See https://github.com/reactphp/http/pull/41#issuecomment-241194727 for reference.)

### CODE REVIEW

**Comment by clue** [comment]
I guess it makes sense to adopt PSR-7's naming convention? http://www.php-fig.org/psr/psr-7/#3-6-psr-http-message-uploadedfileinterface

**Comment by WyriHaximus** [comment]
That makes sense, including method names? But I think actually implementing PSR-7 should be done in a/the next BC release

**Comment by WyriHaximus** [comment]
@clue updated PR to make them resemble PSR-7 UploadedFIle

**Comment by WyriHaximus** [comment]
Ping @clue

File: src/UploadedFile.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace React\Http;
+
+use React\Stream\ReadableStreamInterface;
+
+class UploadedFile implements UploadedFileInterface
```
Comment (clue) [inline src/UploadedFile.php]: Should this be `@internal`?

**Comment by clue** [comment]
This code looks good to me – except that it doesn't have any use case right now :-) IMO it makes sense to also introduce the use case in the same PR as this one would be rather pointless otherwise.

File: src/UploadedFile.php (hunk 1 above)
Comment (WyriHaximus) [inline src/UploadedFile.php]: Yes 👍

**Comment by WyriHaximus** [comment]
Yes it is rather pointless with out a use case. But to avoid creating an enormous PR again I'm going to have a set of PR's depending on each. Those PR's contain small logical components.

**Comment by jsor** [comment]
👍 LGTM

**Comment by clue** [comment]
Didn't mean to close this one, only cleaned up a (seemingly unrelated) feature branch. @WyriHaximus does it make sense to file this again or do you want until the milestone is ready? :+1:

**Comment by WyriHaximus** [comment]
Superseded by #199

**Pull request closed without merging**